### PR TITLE
privilege/privileges: don't reuse chunk in loadTable function

### PR DESCRIPTION
**Pull request by tiancaiamao**
## What have you changed? (mandatory)

Don't reuse chunk in `loadTable` function. 
The memory from the chunk is shallow copy, and `MySQLPrivilege` struct data may be overwritten sometime later, lead to weird TiDB behavior. 

## What are the type of the changes (mandatory)?

- Bug fix (non-breaking change which fixes an issue)

PTAL @XuHuaiyu @zz-jason @coocood

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: privilege/privileges/cache.go [hunk 1]
```diff
@@ -183,16 +183,19 @@ func (p *MySQLPrivilege) loadTable(sctx sessionctx.Context, sql string,
 	defer terror.Call(rs.Close)
 
 	fs := rs.Fields()
-	chk := rs.NewChunk()
-	it := chunk.NewIterator4Chunk(chk)
 	for {
+		// WARNNING: decodeTableRow decodes data from a chunk Row, that is a shallow copy.
```
Comment (zhexuany) [inline privilege/privileges/cache.go]: I hardly saw "WARNNING" in codebase, but `Note` or `NOTE`.

**Comment by tiancaiamao** [comment]
CI is currently down...

**Pull request merged**